### ISSUE TITLE
Ability to set priorityClassName in TektonConfig for resources created by the operator 

### ISSUE THREAD

**Issue by tonygilkerson** [tektoncd/operator#580]
### Feature request

I would like to set the priorityClassName of resources created by the Tekton-Operator, i.e. tekton-dashboard and tekton-pipelines-controller deployments.  It would be nice to enable this in the TektonConfig as shown below in the yaml snipit.

```yaml
kind: TektonConfig
metadata:
  name: config
spec:
  ...
  dashboard:
    priorityClassName: foo
    ...
  pipeline:
    priorityClassName: foo
    ...
  targetNamespace: tekton-pipelines
  trigger:
    priorityClassName: foo
```

### Use case

We need a way to protect core services even when the cluster is over allocated we do this with priority classes.  Also, on our cluster we have several research teams that often want to deploy very large long-running jobs.  We have a method of allowing teams with "credits" to "buy" resources on the system.  The way that we enforce this is by using priorityClasses and let the kubernetes scheduler do it job.   You can think of it as AWS spot instances.  We have a spot priorityClass that can run as long as the system has free resources, but can get preempted if resources are low.  

Therefore if team-a and team-b each have pipelines running in their respective namespaces then I need a way to distinguish the two using priorityClasses

**Comment by nikhil-thomas** [comment]
I think this will be a valuable feature. 🧑‍💻 👍 

@tonygilkerson would a top level `spec.priorityClassName` in TektonConfig (common to all components) good enough. Is the per-component level filed an absolute necessity for your workflow.

We can also think about doing both. ie, a top level default and a per component specification if needed.

```yaml
kind: TektonConfig
metadata:
  name: config
spec:
  ...
  dashboard:
    # priorityClassName defaults to foo from spec.priorityClassName
    ...
  pipeline:
    priorityClassName: foobar
    ...
  targetNamespace: tekton-pipelines
  priorityClassName: foo # top level default
  trigger:
    priorityClassName: foo```

cc @siamaksade @pavol

**Comment by guhaneswaran** [comment]
I would like to work on this issue Can this be assigned to me

**Comment by guhaneswaran** [comment]
> I think this will be a valuable feature. 🧑‍💻 👍
> 
> @tonygilkerson would a top level `spec.priorityClassName` in TektonConfig (common to all components) good enough. Is the per-component level filed an absolute necessity for your workflow.
> 
> We can also think about doing both. ie, a top level default and a per component specification if needed.
> 
> ```yaml
> kind: TektonConfig
> metadata:
>   name: config
> spec:
>   ...
>   dashboard:
>     # priorityClassName defaults to foo from spec.priorityClassName
>     ...
>   pipeline:
>     priorityClassName: foobar
>     ...
>   targetNamespace: tekton-pipelines
>   priorityClassName: foo # top level default
>   trigger:
>     priorityClassName: foo
> ```
> 
> cc @siamaksade

I think `priorityClassName` is a good fit under `spec.config.priorityClassName` instead of `spec.priorityClassName`. Any suggestions ?

**Comment by tonygilkerson** [comment]
putting `priorityClassName` under `spec.config.priorityClassName` works for me

**Comment by vdemeester** [comment]
@sm43 @nikhil-thomas I think it can be closed as #797 got merged 👼🏼
